File: CodesPy/sum/sum_v5.py
#
# v0.2.1 di Luu
#  5 4 5
#
import sys

if len(sys.argv) > 1:
    tot = 0
    try:
        # Convertendo cada argumento para int.
        lista_nums_int = list(map(lambda x: int(x), sys.argv[1:]))
    except ValueError:
        print('Você informou inteiros inválidos.')
        sys.exit(1)
    else:
        print(sum(lista_nums_int))
        sys.exit(0)

cont = 0
numeros = list()
while True:
    # Atribuindo valores do usuário a lista de números.
    try:
        temp_num = int(input(f'Digite o valor do indice {cont} (-1 sair): '))
    except ValueError:
        print('Não é um número inteiro válido.')
        continue
    else:
        if temp_num == -1:
            break
        numeros.append(temp_num)
    
    cont += 1

# Exibindo resultado da soma da lista de números informados.
print(sum(numeros))

File: CodesC/sum/sum_v5.c
/*
    v0.2.1 di Luu
     5 4 5
*/
#include <stdio.h>
#include <stdlib.h>

int sum(int array_nums[], int quant);

char main(int argc, char* argv[])
{
    int quant, i, res, temp_num;
    int tot = 0;

    if (argc > 1)
    {
        // Somando todos os valores passados como argumento.
        for (i = 1; i < argc; i++)
        {
            // Convertendo os valores de argv para int.
            temp_num = atoi(argv[i]);

            // Verificando se é um valor inteiro válido.
            if (temp_num == 0 && argv[i][0] != 0)
            {
                printf("Voce informou valores inteiros invalidos.");
                return 1;
            }

            tot += temp_num;
        }
        printf("%d", tot);

        return 0;
    }

    printf("Quantidade de numeros a serem somados: ");
    scanf("%d", &quant);

    // Criando array com a quantidade de números especificada pelo usuário.
    int numeros[quant];

    // Atribuindo números a cada posição do array.
    for (i = 0; i < quant; i++)
    {
        printf("Digite o valor do indice %d: ", i);
        scanf("%d", &numeros[i]);
    }

    res = sum(numeros, quant);

    printf("A soma de todos os valores e: %d\n", res);
    
    return 0;
}

int sum(int array_nums[], int quant)
{
    int i;
    int tot = 0;

    // Somando todos os valores do array.
    for (i = 0; i < quant; i++)
    {
        tot += array_nums[i];
    }

    return tot;
}